clients/java/src/main/java/io/zeebe/client/impl/worker/JobWorkerImpl.java

```diff
@@ -72,60 +97,99 @@ public boolean isOpen() {
 
   @Override
   public boolean isClosed() {
-    return !isOpen() && jobPoller.get() != null && remainingJobs.get() <= 0;
+    return !isOpen() && claimableJobPoller.get() != null && remainingJobs.get() <= 0;
   }
 
   @Override
   public void close() {
     acquiringJobs.set(false);
   }
 
-  private void tryActivateJobs() {
-    final int remainingJobs = this.remainingJobs.get();
-    if (shouldActivateJobs(remainingJobs)) {
-      activateJobs();
+  /**
+   * Schedules a poll for jobs with a delay of {@code pollInterval}. Does not schedule twice if a
+   * poll is already scheduled.
+   */
+  private void schedulePoll() {
+    if (isPollScheduled.compareAndSet(false, true)) {
+      executor.schedule(this::onScheduledPoll, pollInterval, TimeUnit.MILLISECONDS);
     }
   }
 
-  private void activateJobs() {
-    final JobPoller jobPoller = this.jobPoller.getAndSet(null);
-    if (jobPoller != null) {
-      // check the condition again within the critical section
-      // to avoid race conditions that would let us exceed the buffer size
-      final int currentRemainingJobs = remainingJobs.get();
-      if (shouldActivateJobs(currentRemainingJobs)) {
-        final int maxActivatedJobs = maxJobsActive - currentRemainingJobs;
-        try {
-          jobPoller.poll(
-              maxActivatedJobs,
-              this::submitJob,
-              activatedJobs -> {
-                remainingJobs.addAndGet(activatedJobs);
-                this.jobPoller.set(jobPoller);
-              },
-              this::isOpen);
-        } catch (final Exception e) {
-          LOG.warn("Failed to activate jobs", e);
-          this.jobPoller.set(jobPoller);
-        }
-      } else {
-        this.jobPoller.set(jobPoller);
-      }
+  /** Frees up the scheduler and polls for new jobs. */
+  private void onScheduledPoll() {
+    isPollScheduled.set(false);
+    final int actualRemainingJobs = remainingJobs.get();
+    if (shouldPoll(actualRemainingJobs)) {
+      poll();
     }
   }
 
-  private boolean shouldActivateJobs(final int remainingJobs) {
+  private boolean shouldPoll(final int remainingJobs) {
     return acquiringJobs.get() && remainingJobs <= activationThreshold;
   }
 
-  private void submitJob(final ActivatedJob job) {
-    executor.execute(jobRunnableFactory.create(job, this::jobHandlerFinished));
+  private void poll() {
+    tryClaimJobPoller()
```

Comment:
Let's discuss f2f

Reply:
I've made changes. We now handle all failure cases with backoff (including release of poller and rescheduling poll). I have kept the optional, because I like it. Please let me know if you see any problem with that usage that I seem to have missed.